lib/pages/admin/add_choice_board_category.dart

```diff
@@ -0,0 +1,237 @@
+import 'dart:io';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:seg_coursework_app/pages/admin/admin_choice_boards.dart';
+import 'package:firebase_storage/firebase_storage.dart';
+import 'package:seg_coursework_app/widgets/pick_image_button.dart';
+
+class AddChoiceBoardCategory extends StatefulWidget {
+  const AddChoiceBoardCategory({Key? key}) : super(key: key);
+
+  @override
+  State<AddChoiceBoardCategory> createState() => _AddChoiceBoardCategory();
+}
+
+/// A popup card to add a new category.
+class _AddChoiceBoardCategory extends State<AddChoiceBoardCategory> {
+  File? selectedImage; // hold the currently selected image by the user
+  // controller to retrieve the user input for item name
+  final categoryNameController = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: Padding(
+        padding: const EdgeInsets.all(32.0),
```

Comment:
change the padding value here

Reply:
Change it to what?

Reply:
check the second point of the general comments I wrote. I added this line comment to mark where you should make the change to fix the text field not showing up